src/main/java/blue/Blue.java

```diff
@@ -0,0 +1,138 @@
+package blue;
+
+import java.util.HashMap;
+import java.util.Scanner;
+
+import blue.handler.CommandHandler;
+import blue.handler.DeadlineHandler;
+import blue.handler.DeleteHandler;
+import blue.handler.DoneHandler;
+import blue.handler.EventHandler;
+import blue.handler.ExitHandler;
+import blue.handler.FindHandler;
+import blue.handler.ListHandler;
+import blue.handler.ToDoHandler;
+
+/**
+ * Entry point of the Blue application.
+ * Initializes the application and interacts with the user.
+ */
+public class Blue {
+    private static final String CONFUSED_RESPONSE =
+            "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
+    private HashMap<String, CommandHandler> commandHandlers;
+
+    public Blue() {
+        this("data/tasks.txt");
+    }
+
+    /**
+     * Constructs a Blue instance.
+     *
+     * @param filePath Path to save tasks.
+     */
+    public Blue(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (BlueException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+        initCommandHandlers();
+    }
+
+    /**
+     * Keeps engaging the user until the user input the exit command.
+     */
+    public void run() {
+        ui.showLogo();
+        ui.greet();
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            String input = scanner.nextLine();
+            boolean shouldContinue = canHandle(input);
+            storage.save(tasks);
+            if (!shouldContinue) {
+                break;
+            }
+        }
+        scanner.close();
+    }
+
+    private void initCommandHandlers() {
+        // Construct the handlers
+        ListHandler listHandler = new ListHandler(tasks);
+        ToDoHandler toDoHandler = new ToDoHandler(tasks);
+        DeadlineHandler deadlineHandler = new DeadlineHandler(tasks);
+        EventHandler eventHandler = new EventHandler(tasks);
+        DoneHandler doneHandler = new DoneHandler(tasks);
+        DeleteHandler deleteHandler = new DeleteHandler(tasks);
+        FindHandler findHandler = new FindHandler(tasks);
+        ExitHandler exitHandler = new ExitHandler(tasks);
+
+        // put the handlers into HashMap
+        commandHandlers = new HashMap<>();
+        commandHandlers.put(Command.LIST, listHandler);
+        commandHandlers.put(Command.TODO, toDoHandler);
+        commandHandlers.put(Command.DEADLINE, deadlineHandler);
+        commandHandlers.put(Command.EVENT, eventHandler);
+        commandHandlers.put(Command.DONE, doneHandler);
+        commandHandlers.put(Command.DELETE, deleteHandler);
+        commandHandlers.put(Command.FIND, findHandler);
+        commandHandlers.put(Command.EXIT, exitHandler);
+    }
+
+    private boolean canHandle(String input) {
```

Comment:
This method name sounds, to me, like a boolean rather than a verb. Perhaps a suggestion eg. `checkCanHandle`

Reply:
I am actually not quite sure about how to name this method. `canHandle` sounds like only checking whether it can handle, but it is actually also handling it, which makes the name misleading.

The `boolean` was needed to determine whether it is time to exit the application. I am not sure whether it is better to handle the exit command at the GUi level or at the level of the bot logic. I might improve the design if I found a better way.